tests/integ/test_dataframe.py

```diff
@@ -1475,14 +1475,67 @@ def test_write_temp_table(session, save_mode):
     table_name = Utils.random_name_for_temp_object(TempObjectType.TABLE)
     df = session.create_dataframe([(1, 2), (3, 4)]).toDF("a", "b")
     try:
-        df.write.save_as_table(table_name, mode=save_mode, create_temp_table=True)
+        df.write.save_as_table(table_name, mode=save_mode, table_type="temp")
         Utils.check_answer(session.table(table_name), df, True)
         table_info = session.sql(f"show tables like '{table_name}'").collect()
         assert table_info[0]["kind"] == "TEMPORARY"
     finally:
         Utils.drop_table(session, table_name)
 
 
+@pytest.mark.parametrize(
+    "save_mode", ["append", "overwrite", "ignore", "errorifexists"]
+)
+def test_write_temporary_table(session, save_mode):
+    table_name = Utils.random_name_for_temp_object(TempObjectType.TABLE)
+    df = session.create_dataframe([(1, 2), (3, 4)]).toDF("a", "b")
+    try:
+        df.write.save_as_table(table_name, mode=save_mode, table_type="temporary")
+        Utils.check_answer(session.table(table_name), df, True)
+        table_info = session.sql(f"show tables like '{table_name}'").collect()
+        assert table_info[0]["kind"] == "TEMPORARY"
+    finally:
+        Utils.drop_table(session, table_name)
+
+
+@pytest.mark.parametrize(
+    "save_mode", ["append", "overwrite", "ignore", "errorifexists"]
+)
+def test_write_temp_table_no_breaking_change(session, save_mode):
+    table_name = Utils.random_name_for_temp_object(TempObjectType.TABLE)
+    df = session.create_dataframe([(1, 2), (3, 4)]).toDF("a", "b")
+    try:
+        with pytest.deprecated_call(match="create_temp_table is deprecated"):
+            df.write.save_as_table(table_name, mode=save_mode, create_temp_table=True)
+        Utils.check_answer(session.table(table_name), df, True)
+        table_info = session.sql(f"show tables like '{table_name}'").collect()
+        assert table_info[0]["kind"] == "TEMPORARY"
+    finally:
+        Utils.drop_table(session, table_name)
+
+
+@pytest.mark.parametrize(
+    "save_mode", ["append", "overwrite", "ignore", "errorifexists"]
+)
+def test_write_transient_table(session, save_mode):
```

Comment:
Use `pytest.mark.parameterize` to have an argument `table_type` so we don't need to duplicate this test?